hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.controller.js

```diff
@@ -381,6 +412,35 @@ define([
           $rootScope.options.pay[optionName] = _.mapValues(_.keyBy(data, 'value'), 'label');
         });
     }
+
+    /**
+     * Loads updated record of providers option group
+     *
+     * @param {Array} providers
+     * @returns {Promise}
+     */
+    function loadProviderOptions (providers) {
+      var providersList = [];
+      var optionGroup = '';
+
+      return utilsService.getOptionValues(providers)
+        .then(function (data) {
+          providers.forEach(function (provider) {
```

Comment:
this function is too long and complex, extract what is going on here and put it in a separate function and document it. In the end it should look like this
```js
return utilsService.getOptionValues(providers)
  .then(processProvidersApiResult)  // name of function is just a suggestion
  .then(function (providers) {
    $scope.providers = providers;
  });